src/test/shell/bazel/cc_integration_test.sh

```diff
@@ -1655,6 +1655,120 @@ EOF
     @other_repo//pkg:test &>"$TEST_log" || fail "Test should succeed"
   expect_log "in external/other_repo/pkg/test.cpp: 'other_repo'"
   expect_log "in pkg/library.cpp: ''"
+
+  # test Runfiles::CurrentRepository()
+  cat > pkg/BUILD.bazel <<'EOF'
+cc_library(
+  name = "library",
+  hdrs = ["library.h"],
+  deps = ["@bazel_tools//tools/cpp/runfiles", ":gen_library", "@other_repo//gen_pkg:gen_library"],
+  visibility = ["//visibility:public"],
+)
+
+genrule(
+  name = "gen_library_hdr",
+  outs = ["gen_library.h"],
+  cmd = """
+cat >$@ <<EOF_
+#include <iostream>
+
+#include "tools/cpp/runfiles/runfiles.h"
+
+using bazel::tools::cpp::runfiles::Runfiles;
+
+inline void gen_print_repo_name() {
+  std::cout << "in " << __FILE__ << ": '" << Runfiles::CurrentRepository() << "'" << std::endl;
+}
+EOF_
+"""
+)
+
+cc_library(
+  name = "gen_library",
+  hdrs = ["gen_library.h"],
+  deps = ["@bazel_tools//tools/cpp/runfiles"],
+)
+
+cc_binary(
+  name = "binary",
+  srcs = ["binary.cpp"],
+  deps = [":library"],
+)
+
+cc_test(
+  name = "test",
+  srcs = ["test.cpp"],
+  deps = [":library"],
+)
+EOF
+
+  mkdir -p other_repo/gen_pkg
+  cat > other_repo/gen_pkg/BUILD.bazel <<'EOF'
+genrule(
+  name = "gen_library_hdr",
+  outs = ["gen_library.h"],
+  cmd = """
+cat >$@ <<EOF_
+#include <iostream>
+
+#include "tools/cpp/runfiles/runfiles.h"
+
+using bazel::tools::cpp::runfiles::Runfiles;
+
+inline void other_repo_gen_print_repo_name() {
+  std::cout << "in " << __FILE__ << ": '" << Runfiles::CurrentRepository() << "'" << std::endl;
+}
+EOF_
+"""
+)
+
+cc_library(
+  name = "gen_library",
+  hdrs = ["gen_library.h"],
+  deps = ["@bazel_tools//tools/cpp/runfiles"],
+  visibility = ["//visibility:public"],
+)
+EOF
+
+  cat > pkg/library.h <<'EOF'
+#include <iostream>
+#include "tools/cpp/runfiles/runfiles.h"
+#include "pkg/gen_library.h"
+#include "gen_pkg/gen_library.h"
+
+using bazel::tools::cpp::runfiles::Runfiles;
+
+inline void print_repo_name() {
+  std::cout << "in " << __FILE__ << ": '" << Runfiles::CurrentRepository() << "'" << std::endl;
+  gen_print_repo_name();
+  other_repo_gen_print_repo_name();
+}
+EOF
+
+  bazel run //pkg:binary &>"$TEST_log" || fail "Run should succeed"
```

Comment:
I think that the test fails because Bazel uses C++11 by default. You probably have to add `--copt=c++20`.

Reply:
We could, but I think that the test just wouldn't run if we did: Since Bazel passes `--std=0x` on Unix by default, it just won't be available on any of the runners.